### PR TITLE
Implement get appointment by date feature

### PR DESCRIPTION
**Pull request by PeiYee88**
Command format:
`get /apptbd [DATE]`
Example:
`get /apptbd 21-12-2002`

### CODE REVIEW

File: src/main/java/seedu/address/model/person/AppointmentByDatePredicate.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public boolean test(Person person) {
+        Boolean pastAppointments = appointments.stream().anyMatch(keyword -> person.getPastAppointments().stream()
+                .anyMatch(appointment -> keyword.compareTo(appointment.getDate()) == 0));
+        Boolean upcomingAppointments = appointments.stream().anyMatch(keyword
+                -> person.getUpcomingAppointment().isPresent())
+                && appointments.stream().anyMatch(keyword -> person.getUpcomingAppointment().stream()
+                .anyMatch(appointment -> keyword.compareTo(appointment.getDate()) == 0));
+        return pastAppointments || upcomingAppointments;
+    }
```
Comment (nehcuy) [inline src/main/java/seedu/address/model/person/AppointmentByDatePredicate.java:35]: Great implementation of the `test` method! Perhaps better abstraction for the booleans within `upcomingAppointments` could have been better. For instance, the `upcomingAppointments` variable could be further split into `Boolean hasUpcomingAppt` && `Boolean isSameDate`.

**Review by nehcuy — APPROVED** [review]
LGTM! Perhaps you could fix the variable names in a future PR :) Thanks!

File: src/main/java/seedu/address/logic/commands/getcommands/GetAppointmentByDateCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.person.AppointmentByDatePredicate;
+
+/**
+ * Finds and lists all patients on the same appointment date.
+ *
+ */
+public class GetAppointmentByDateCommand extends GetCommand {
+
+    public static final String APPOINTMENT_BY_DATE_PREFIX = "/apptbd";
```
Comment (kxrt) [inline src/main/java/seedu/address/logic/commands/getcommands/GetAppointmentByDateCommand.java:17]: Would it be possible to change this prefix to `/appton`? Sounds better and is easier for users to read.

**Review by kxrt — CHANGES_REQUESTED** [review]
LGTM, just needs a tiny change to the prefix!

File: src/main/java/seedu/address/logic/commands/getcommands/GetAppointmentByDateCommand.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
  */
 public class GetAppointmentByDateCommand extends GetCommand {
 
-    public static final String APPOINTMENT_BY_DATE_PREFIX = "/apptbd";
```
Comment (kxrt) [inline src/main/java/seedu/address/logic/commands/getcommands/GetAppointmentByDateCommand.java]: Looks good, thanks for changing this!

**Review by kxrt — APPROVED** [review]

**Pull request merged**